module-network.tf

```diff
@@ -83,6 +83,7 @@ module "network" {
   create_bastion               = var.create_bastion
   create_fss                   = var.create_fss
   create_nsgs                  = var.create_nsgs
```

Comment:
what's the difference between `create_nsgs` and `create_nsgs_always`?

Reply:
This is similar to the subnets `never`/`auto`/`always` use-case, and I added a comment that we might consider aligning on that for NSGs.

NSGs are created by default when called for, e.g. operator NSG when operator is enabled, Pods with NPN cluster, etc. In cases where network resources are created separately and re-used, `create_nsgs_always` forces creation even when the related components are disabled.